server/services/voip/service.ts

```diff
@@ -126,39 +127,39 @@ export class VoipService extends ServiceClass implements IVoipService {
 		return this.commandHandler.executeCommand(Commands.queue_summary);
 	}
 
-	async getQueuedCallsForThisExtension(requestParams: { extension: string }): Promise<IVoipConnectorResult> {
+	async getQueuedCallsForThisExtension({ extension }: { extension: string }): Promise<IVoipConnectorResult> {
 		const membershipDetails: IQueueMembershipDetails = {
 			queueCount: 0,
 			callWaitingCount: 0,
-			extension: '',
+			extension,
 		};
-		membershipDetails.extension = requestParams.extension;
-		const queueSummary = Promise.await(this.commandHandler.executeCommand(Commands.queue_summary)) as IVoipConnectorResult;
-		for (const queue of queueSummary.result as IQueueSummary[]) {
-			const queueDetails = Promise.await(
-				this.commandHandler.executeCommand(Commands.queue_details, { queueName: queue.name }),
-			) as IVoipConnectorResult;
-			this.logger.debug({ msg: 'API = voip/queues.getCallWaitingInQueuesForThisExtension queue details = ', result: queueDetails });
-			if (!(queueDetails.result as unknown as IQueueDetails).members) {
+		const queueSummary = (await this.commandHandler.executeCommand(Commands.queue_summary)) as IVoipConnectorResult;
```

Comment:
Queue membership wont change too often.I can imagine it to be changing (A safest measure) once a day. In reality it will not change for weeks or months once the initial configuration is done. But Queue summary and queue details also contain some other more dynamic information which we will need. e.g available agents, talk time for the queue, longest hold time. There are few ways I think we can use for optimisation
1. Memoize the part that does not change. Change the APIs in such a way that if slow changing information is asked for, we give it from the memory.
2. We provide a different set of APIs for getting dynamically changing parameter. (Same as above but slightly altered.)

I think we should keep it as is for MVP and then look at the optimisation part. This will also help gathering the customer data on the usage. Furthermore, we should configure the memoization parameters per command.

Reply:
Yup agreed, should be after-mvp. But yeah, if won't change too often, a memoize would come handy, and we can have both, a memoized and a direct call depending on how we plan to use it 😬